h5pt.go

```diff
@@ -82,170 +97,143 @@ func (t *Table) ReadPackets(start, nrecords int, data interface{}) error {
-		cData = unsafe.Pointer(&stringData)
+		ad.strs = append(ad.strs, unsafe.Pointer(stringData))
+		dataPtr = unsafe.Pointer(&stringData)
+		dataSize = C.size_t(unsafe.Sizeof(dataPtr))
 
 	case reflect.Ptr:
-		ptrVal := rp.Elem()
-		cData = unsafe.Pointer(&ptrVal)
-
-	case reflect.Bool:
-		val := C.uchar(0)
-		if data.(bool) {
-			val = 1
-		}
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		return extractValues(ad, rv.Elem())
 
 	case reflect.Int8:
-		val := C.int8_t(data.(int8))
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		val := C.int8_t(rv.Int())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 1
 
 	case reflect.Uint8:
-		val := C.uint8_t(data.(uint8))
-		cData = unsafe.Pointer(&val)
-
-	case reflect.Uint16:
-		val := C.uint16_t(data.(uint16))
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		val := C.uint8_t(rv.Uint())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 1
 
 	case reflect.Int16:
-		val := C.int16_t(data.(int16))
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		val := C.int16_t(rv.Int())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 2
+
+	case reflect.Uint16:
+		ad.numRecords = 1
+		val := C.uint16_t(rv.Uint())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 2
 
 	case reflect.Int32:
-		val := C.int32_t(data.(int32))
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		val := C.int32_t(rv.Int())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 4
 
 	case reflect.Uint32:
-		val := C.uint32_t(data.(uint32))
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		val := C.uint32_t(rv.Uint())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 4
 
 	case reflect.Int64:
-		val := C.int64_t(data.(int64))
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		val := C.int64_t(rv.Int())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 8
 
 	case reflect.Uint64:
-		val := C.uint64_t(data.(uint64))
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		val := C.uint64_t(rv.Uint())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 8
 
 	case reflect.Float32:
-		val := C.float(data.(float32))
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		val := C.float(rv.Float())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 4
 
 	case reflect.Float64:
-		val := C.double(data.(float64))
-		cData = unsafe.Pointer(&val)
+		ad.numRecords = 1
+		val := C.double(rv.Float())
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 8
+
+	case reflect.Bool:
+		ad.numRecords = 1
+		val := C.uchar(0)
+		if rv.Bool() {
+			val = 1
+		}
+		dataPtr = unsafe.Pointer(&val)
+		dataSize = 1
 
 	default:
-		return fmt.Errorf("hdf5: PT Append does not support datatype (%s).", rt.Kind())
+		err = fmt.Errorf("hdf5: PT Append does not support datatype (%s).", rt.Kind())
+		return
+	}
+
+	ad.memSize = dataSize + ad.offset
+	ad.ptr = C.realloc(ad.ptr, ad.memSize)
+	C.memcpy(unsafe.Pointer(uintptr(ad.ptr)+uintptr(ad.offset)), dataPtr, dataSize)
+	ad.offset += dataSize
+
+	return
+}
+
+// Append appends packets to the end of a packet table.
```

Comment:
Add comment similar to the one above about visibility of fields and remove the named return.